### ISSUE TITLE
no `comparison-overlap` error when using `in` operator on `list`

### ISSUE THREAD

**Issue by DetachHead** [python/mypy#11425]
```py
foo: list[str] = []

print(1 in foo)  # no error
```
broken in 0.920+0ec868fe778c57b52f6df54f54eb19616e1d6ba2

works in 0.910: https://mypy-play.net/?mypy=latest&python=3.10&flags=show-column-numbers%2Cshow-error-codes%2Cstrict%2Ccheck-untyped-defs%2Cdisallow-any-decorated%2Cdisallow-any-expr%2Cdisallow-any-explicit%2Cdisallow-any-generics%2Cdisallow-any-unimported%2Cdisallow-incomplete-defs%2Cdisallow-subclassing-any%2Cdisallow-untyped-calls%2Cdisallow-untyped-decorators%2Cdisallow-untyped-defs%2Cwarn-incomplete-stub%2Cwarn-redundant-casts%2Cwarn-return-any%2Cwarn-unreachable%2Cwarn-unused-configs%2Cwarn-unused-ignores&gist=405fae340ebf7e84560dd0b29cbf18b4

**Comment by sobolevn** [comment]
I think that we can use `__getitem__` and `__contains__` magic methods 🤔

**Comment by sobolevn** [comment]
Ok, this is not really a mypy issue. Mypy works correctly for types where `__contains__` definition is correct. https://github.com/python/mypy/blob/e324460c73404fcfb19e920cdb6f415979fb803e/mypy/checkexpr.py#L2235-L2264

I think that you should report this here: https://github.com/python/typeshed/blob/e2ee7c297dd33aac2035b572fc88e7471118ec94/stdlib/builtins.pyi#L783

**Comment by erictraut** [comment]
It appears this bug no longer exists in mypy 1.5.

**Comment by DetachHead** [comment]
maybe a test should be added to make sure this bug doesn't come back

**Comment by hauntsaninja** [comment]
I think this bug report is buggy. As far as I can tell, mypy has always complained about this, in the specific hash you mention, 0.910, 0.920 and 0.921.

**Comment by DetachHead** [comment]
no idea how that happened. i just checked that commit and can't reproduce it anymore either which is strange because it was definitely broken when i raised it
